lib/internal/promiseCallback.js

```diff
@@ -0,0 +1,24 @@
+import noop from 'lodash/noop';
+
+var supportsPromise = typeof Promise === 'function';
```

Comment:
A pretty crude way to detect promise support.  This would be a 3.0 feature.  I thinking of dropping support for non-ES2015 environments, in which case we could always return a Promise.

Reply:
:+1: for dropping pre-ES2015 support in v3.